### PR TITLE
Update for MinimalReferenceExpressionsVisitor

### PR DESCRIPTION
**Pull request by apmoriarty**
Update for MinimalReferenceExpressionsVisitor to handle special case for assignment nodes.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java [hunk 1]
```diff
@@ -126,7 +127,10 @@ public Object visit(ASTAndNode node, Object data) {
         if (!isValid)
             return data;
         
-        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
+        JexlASTHelper.validateLineage(node, true);
+        QueryPropertyMarker.Instance instance = QueryPropertyMarkerVisitor.getCopiedInstance(node);
+        JexlASTHelper.validateLineage(node, true);
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java]: Is this not overkill?

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java [hunk 2]
```diff
@@ -169,11 +173,11 @@ private boolean isAnyChildAWrappedSingleTerm(JexlNode node) {
     
     private boolean isWrappedSingleTerm(JexlNode node) {
         if (isParen(node)) {
-            boolean isMarkerNode = QueryPropertyMarker.findInstance(node).isAnyType();
+            boolean isMarkerNode = QueryPropertyMarkerVisitor.getCopiedInstance(node).isAnyType();
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java]: why copy?

**Review by ivakegg — CHANGES_REQUESTED** [review]
I don't quite understand purpose of copying.

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java]: I'll remove that, it's a vestige from testing

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/MinimalReferenceExpressionsVisitor.java]: That was a workaround until https://github.com/NationalSecurityAgency/datawave/pull/1297 could get merged. My preference is to get that in first, and then update this merge request

**Review by ivakegg — APPROVED** [review]

**Review by jschmidt10 — APPROVED** [review]

**Pull request merged**